src/absy.rs

```diff
@@ -288,19 +289,11 @@ impl<T: Field> Expression<T> {
                         }
                         assert_eq!(num, T::zero());
                     } else {
-                        println!(
-                            "Could not calculate variable {:?}, inputs: {:?}",
+                        panic!(
```

Comment:
Theoretically we could remove this part as semantic checks ensure all variables are defined and the CLI ensures all main arguments are specified. Thoughts?